File: src/1137_nth_tribonacci_number/NthTribonacciNumber.java
public class NthTribonacciNumber {
    public int tribonacci(int n) {
        if (n == 0)
            return 0;
        if (n == 1 || n == 2)
            return 1;
        int a = 0, b = 1, c = 1;
        for (int i = 3; i <= n; ++i) {
            int next = a + b + c;
            a = b;
            b = c;
            c = next;
        }
        return c;
    }

    // Test scaffolding
    public static void main(String[] args) {
        NthTribonacciNumber sol = new NthTribonacciNumber();
        System.out.println(sol.tribonacci(0)); // 0
        System.out.println(sol.tribonacci(1)); // 1
        System.out.println(sol.tribonacci(2)); // 1
        System.out.println(sol.tribonacci(3)); // 2
        System.out.println(sol.tribonacci(4)); // 4
        System.out.println(sol.tribonacci(25)); // 1389537
    }
}

File: src/1137_nth_tribonacci_number/nth_tribonacci_number.cpp
#include <iostream>
using namespace std;

class Solution {
   public:
    int tribonacci(int n) {
        if (n == 0) return 0;
        if (n == 1 || n == 2) return 1;
        int a = 0, b = 1, c = 1;
        for (int i = 3; i <= n; ++i) {
            int next = a + b + c;
            a = b;
            b = c;
            c = next;
        }
        return c;
    }
};

// Test scaffolding
int main() {
    Solution sol;
    cout << sol.tribonacci(0) << endl;   // 0
    cout << sol.tribonacci(1) << endl;   // 1
    cout << sol.tribonacci(2) << endl;   // 1
    cout << sol.tribonacci(3) << endl;   // 2
    cout << sol.tribonacci(4) << endl;   // 4
    cout << sol.tribonacci(25) << endl;  // 1389537
    return 0;
}
